addons/purchase_stock/models/stock_move.py

```diff
@@ -212,3 +213,16 @@ def _get_all_related_aml(self):
 
     def _get_all_related_sm(self, product):
         return super()._get_all_related_sm(product) | self.filtered(lambda m: m.purchase_line_id.product_id == product)
+
+    def _get_purchase_line_and_partner_from_chain(self):
+        moves_to_check = deque(self)
+        seen_moves = set()
+        while moves_to_check:
+            current_move = moves_to_check.popleft()
+            if current_move.purchase_line_id:
+                return current_move.purchase_line_id.id, current_move.picking_id.partner_id.id
+            seen_moves.append(current_move)
```

Comment:
this would have made the runbot red (should be `add`), if a test called it, which would be nice to have :)

Reply:
I added a test for this behaviour in test_create_picking.py in purchase_stock